jobs/otel-collector-windows/templates/config.yml.erb

```diff
@@ -1,35 +1,50 @@
 <%=
 def config
   @config ||= begin
-    cfg = retrieve_config
+    cfg = retrieve_property('config')
+    secrets = retrieve_property('secrets').to_h { |x| [x['name'], { 'cert' => x['cert'], 'key' => x['key'], 'ca' => x['ca'], 'secret' => x['secret'] }] }
+    used_secrets = {}
+    cfg = interpolate_secrets(cfg, secrets, used_secrets)
```

Comment:
It seems like interpolate_secrets is modifying cfg, so maybe this should be interpolate_secrets! and not assigning to config. Admittedly it's a little confusing since there isn't an object to call the method on so it would look like the line below where the bang means it might raise an error